src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -0,0 +1,160 @@
+  // May limit how much we can do in terms of JUnit tests
+  //Base arm PID values
+  private double BP = 1;
+  private double BI = 0;
+  private double BD = 0.1;
+  //Elbow arm PID values
+  private double EP = 0.05;
+  private double EI = 0;
+  private double ED = 0.2;
+  private CANSparkMax motorBaseOne;
+  private CANSparkMax motorBaseTwo;
+  private CANSparkMax motorElbow;
+  private final Encoder motorBaseOneEncoder;
+  private final EncoderSim motorBaseOneEncoderSim;
+  private final Encoder motorElbowEncoder;
+  private final EncoderSim motorElbowEncoderSim;
+  private final PIDController basePIDController;
+  private final PIDController elbowPIDController;
+  private ShuffleboardTab configTab = Shuffleboard.getTab("Arm");
+  //Base Arm Values for Simulation
+  private GenericEntry simBArmAngle = configTab.add("SimBase Arm Angle", 0.0).getEntry();
+  private GenericEntry simBEncoderPos = configTab.add("SimBase Encoder Angle", 0.0).getEntry();
+  private GenericEntry simBOutSet = configTab.add("SimBase Output", 0.0).getEntry();
+  private GenericEntry simBError = configTab.add("SimBase Error", 0.0).getEntry();
+  private GenericEntry simBVoltage = configTab.add("SimBase Motor Voltage", 0.0).getEntry();
+  //Elbow Arm Values for Simulation
+  private GenericEntry simEArmAngle = configTab.add("SimElbow Arm Angle", 0.0).getEntry();
+  private GenericEntry simEEncoderPos = configTab.add("SimElbow Encoder Angle", 0.0).getEntry();
+  private GenericEntry simEOutSet = configTab.add("SimElbow Output", 0.0).getEntry();
+  private GenericEntry simEError = configTab.add("SimElbow Error", 0.0).getEntry();
+  private GenericEntry simEVoltage = configTab.add("SimElbow Motor Voltage", 0.0).getEntry();
+  
+
+  // SIM VALUES:
+  private double baseGearing = 15.;
+  private double elbowGearing = 8.57142857;
+  public static double baseArmLength = Units.inchesToMeters(43.5);
+  public static double elbowArmLength = Units.inchesToMeters(37.5);
+  private double minBAngle = Units.degreesToRadians(45);
+  private double maxBAngle = Units.degreesToRadians(91);
+  private double minEAngle = Units.degreesToRadians(-70);
+  private double maxEAngle = Units.degreesToRadians(89);
+  private double baseArmMass = Units.lbsToKilograms(15); //15
+  private double elbowArmMass = Units.lbsToKilograms(15);
+  private final double baseArmInertia = SingleJointedArmSim.estimateMOI(baseArmLength, baseArmMass);
+  private final double elbowArmInertia = SingleJointedArmSim.estimateMOI(elbowArmLength, elbowArmMass);
+  private double simBCurrentAngle;
+  private double simECurrentAngle;
+
+  private final SingleJointedArmSim baseArmSim = new SingleJointedArmSim(DCMotor.getNEO(2), baseGearing, baseArmInertia, baseArmLength, minBAngle, maxBAngle, baseArmMass, false);
+  private final SingleJointedArmSim elbowArmSim = new SingleJointedArmSim(DCMotor.getNEO(1), elbowGearing, elbowArmInertia, elbowArmLength, minEAngle, maxEAngle, elbowArmMass, false);
+
+  public ArmSubsystem() {
```

Comment:
I don't think you are taking into account the initial angles of the two arms. The encoders will start at zeros, but the angles will not be zero. Make sure compensate for that initial offset

Reply:
done, encoders are now set to arm minimum angle values at the beginning of simulation